File: VisualStudio/VBasic/w7cpframe/clsPfForce.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On

Public Class clsPfForce

#Const isUsingExcel = 0

    Public axial As Double       '.. axial force ..
    Public shear As Double       '.. shear force ..
    Public momnt As Double      '.. end moment ..


    Sub initialise()
        axial = 0
        shear = 0
        momnt = 0
    End Sub

    Sub cprint()
        Console.WriteLine(format(axial,"0.00") + " : " + format(shear,"0.00") +  " : " + format( momnt,"0.00"))
    End Sub

#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 1).Value = axial
            .Offset(RecNum, 2).Value = shear
            .Offset(RecNum, 3).Value = momnt
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            axial = .Offset(RecNum, 1).Value
            shear = .Offset(RecNum, 2).Value
            momnt = .Offset(RecNum, 3).Value
        End With
    End Sub

#End If

End Class

File: Eclipse/w7cpframe/src/clsPfForce.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//

//import java.io.*;


    class clsPfForce
    {


        public double axial = 0;        //.. axial force ..
        public double shear = 0;        //.. shear force ..
        public double momnt = 0;         //.. end moment ..



        public void initialise()
        {
            axial = 0;
            shear = 0;
            momnt = 0;
        }


        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%12.48f",axial);
            s = s + String.format("%12.48f",shear);
            s = s + String.format("%12.48f",momnt);

            return s;

        }



        public void cprint()
        {
            System.out.println(sprint());
        }




    } //class

